src/InputField.js

```diff
@@ -9,72 +9,144 @@ import Label from './Label'
 import Input from './Input'
 import theme from './theme'
 
-const InfoBox = styled(Box)`
-  border-width: 1px;
-  border-radius: 0 0 2px 2px;
-  border-color: ${({ color }) => color};
-  border-style: solid;
-  box-shadow: 0 -1px 0 0 ${({ color }) => color};
-`
+class InputField extends React.Component {
+  render() {
+    const { color, label, icon, children, ...props } = this.props
 
-const InputField = ({ id, label, color, icon, info, ...props }) => {
-  return (
-    <div>
-      <Label htmlFor={id} pl={13}>
-        {label}
-      </Label>
-      <Flex align="center" width={1}>
-        <Input
-          id={id}
-          pt={20}
-          pb={8}
-          mt={-20}
-          width={1}
-          color={color}
-          pr={icon && 40}
-          aria-describedby={info && `${id}-info`}
-          {...props}
-        />
-        {icon && (
-          <Box ml={-40} mr={14} mt={-14}>
-            <Icon name={icon} color={color} />
-          </Box>
-        )}
-      </Flex>
-      {info && (
-        <InfoBox
-          width={1}
-          mb={-24}
-          color={theme.colors[color] || theme.colors.borderGray}
-          bg={theme.colors[color] || theme.colors.borderGray}
-        >
-          <Text
-            id={`${id}-info`}
-            bold
-            fontSize={10}
-            px={13}
-            py={1}
-            color={'white'}
-          >
-            {info}
-          </Text>
-        </InfoBox>
-      )}
-    </div>
-  )
+    let InputChild
+    let inputPosition = -1
+    let LabelChild
+    let BeforeIcon
+    let AfterIcon
+    let inputId
+    let iconAdjustment
+
+    React.Children.forEach(children, (child, index) => {
+      if (child) {
+        if (child.type === Label) {
+          LabelChild = child
+        }
+        if (child.type === Input) {
+          inputPosition = index
+          InputChild = child
+          inputId = child.props.id
+        }
+        if (child.type === Icon) {
+          if (inputPosition < 0) {
+            BeforeIcon = child
+            iconAdjustment = child.props.size - Icon.defaultProps.size
+          } else {
+            AfterIcon = child
+          }
+        }
+      }
+    })
+
+    // Handle old version on component's api
+    if (icon) {
+      AfterIcon = <Icon name={icon} />
+    }
+    if (label) {
+      LabelChild = <Label>{label}</Label>
+    }
+    if (!InputChild) {
+      InputChild = <Input />
+    }
+
+    return (
+      <Box>
+        {LabelChild &&
```

Comment:
We should probably show the label if `InputChild.props.value` contains a value. So that we hide the label if the input is not filled. It would also be awesome to add that nice `transition: padding .2s ease-in-out;` to slide the text down when the user start typing.